setup.py

```diff
@@ -1,6 +1,20 @@
+from jupyter_packaging import wrap_installers, npm_builder
 from setuptools import find_packages, setup
 from distutils.util import convert_path
-import subprocess
+import os.path
+
+HERE = os.path.abspath(os.path.dirname(__file__))
+
+# Representative files that should exist after a successful build
+jstargets = [
+    os.path.join(HERE, "nbgitpuller", "static", "dist", "bundle.js"),
+]
+
+# https://github.com/jupyter/jupyter-packaging/blob/0.10.4/README.md#as-a-build-requirement
+jsdeps = npm_builder(build_cmd="webpack")
+cmdclass = wrap_installers(
+    pre_develop=jsdeps, pre_dist=jsdeps,
+    ensured_targets=jstargets, skip_if_exists=jstargets)
```

Comment:
I think you should wrap this in a try catch block, so that you don't assume jupyter_packaging to be available.

See this example from the jupyter_packaging readme file https://github.com/jupyter/jupyter-packaging#as-a-build-requirement

Reply:
I originally followed the example, but that means setup.py silently ignores the js build which didn't seem particularly great.

It looks like `pyproject.toml` is included anyway in the sdist (at least it is when I tested locally, and `pip install ~/path/to/nbgitpuller/dist/nbgitpuller-1.0.1.dev0.tar.gz` works without errors).

What do you think?

Reply:
Long discussion here which I don't fully understand https://github.com/pypa/setuptools/pull/1650
... but it looks like this project assumes [pep517 anyway](https://github.com/pypa/build)
https://github.com/jupyterhub/nbgitpuller/blob/4ffd0097a6971f1f2c62821079051a6fb30830ae/.github/workflows/publish.yml#L27

Reply:
Nice investigation @manics! Okay my understanding now:

1. pyproject.toml will always ship with the source when we package the source with `build --sdist --wheel .` This means that `jupyter_packaging` can be assumed in setup.py, so we don't need to wrap the import statement etc in a try block.
2. The nbgitpuller/static/dist content will also be shipped, and jupyter_packaging will not attempt to do a npm setup unless bundle.js isn't around anyhow because of `skip_if_exists=jstargets`. This means that we currently can do `pip install .` from the source file. If you on the other hand wants to rebuild the bundle.js file you won't be able to do so because then we miss package.json. I've verified this part.

I suggest we include package.json in our MANIFEST.in file as part of this PR.

I can't make a convenient code suggestion about that so I pushed a commit. If it LGTY also @manics then go for a merge! This LGTM now!